core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverAndListenerJobFactory.java

```diff
@@ -0,0 +1,41 @@
+package ai.timefold.solver.core.api.solver;
+
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+
+/**
+ * Provides a fluid contract that allows customization and submission of planning problems to solve. It differs {@link SolverJobFactory}
+ * because it allows to listen to intermediate best solutions.
+ * <p>
+ * To create a {@link SolverJob}, use {@link #solveAndListen()}.
+ *
+ * @param <Solution_>  the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <ProblemId_> the ID type of a submitted problem, such as {@link Long} or {@link UUID}.
+ */
+public interface SolverAndListenerJobFactory<Solution_, ProblemId_> extends SolverJobFactory<Solution_, ProblemId_> {
+
+    // ************************************************************************
+    // With methods
+    // ************************************************************************
+    @Override
+    SolverAndListenerJobFactory<Solution_, ProblemId_> withFinalBestSolutionConsumer(Consumer<? super Solution_> finalBestSolutionConsumer);
+
+    @Override
+    SolverAndListenerJobFactory<Solution_, ProblemId_> withExceptionHandler(BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler);
+
+    @Override
+    SolverAndListenerJobFactory<Solution_, ProblemId_> withTerminationConfig(TerminationConfig terminationConfig);
```

Comment:
Please replace this with a method accepting a `SolverConfigOverride`, which will have this method on it.
The reason is that we will then use the same override in both the `Solver` and the `SolverManager` builders, without having to copy each of the methods.

Reply:
Let's wait for the design meeting to define the next steps here.

Reply:
What do you think about hiding the config override class from the user? 
I suggest simplifying the user's experience by only dealing with the Termination configuration.

Reply:
Let's expose the override for now. We can always simplify later.